File: src/main/java/com/lcaohoanq/nocket/domain/reaction/PostReactionPort.java
package com.lcaohoanq.nocket.domain.reaction;

import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.lcaohoanq.nocket.domain.post.PostPort.PostResponse;
import com.lcaohoanq.nocket.domain.reaction.ReactionPort.ReactionResponse;
import com.lcaohoanq.nocket.domain.user.UserPort.UserResponse;
import java.util.UUID;

public interface PostReactionPort {

    @JsonPropertyOrder(
        {
            "id",
            "user",
            "post",
            "reaction"
        }
    )
    record PostReactionResponse(
        UUID id,
        UserResponse user,
        PostResponse post,
        ReactionResponse reaction
    ){}
    
}

File: src/main/kotlin/com/lcaohoanq/nocket/domain/reaction/PostReactionPort.kt
package com.lcaohoanq.nocket.domain.reaction

import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.lcaohoanq.nocket.domain.post.PostPort.PostResponse
import com.lcaohoanq.nocket.domain.reaction.ReactionPort.ReactionResponse
import com.lcaohoanq.nocket.domain.user.UserPort.UserResponse
import java.util.*

interface PostReactionPort {
    @JsonPropertyOrder(
        "id", "user", "post", "reaction"
    )
    data class PostReactionResponse(
        val id: UUID,
        val user: UserResponse,
        val post: PostResponse,
        val reaction: ReactionResponse
    )
}
